Harvest.Core/Services/InvoiceService.cs

```diff
@@ -25,57 +25,74 @@ public class InvoiceService : IInvoiceService
+            _emailService = emailService;
         }
         public async Task<Result<int>> CreateInvoice(int projectId, bool manualOverride = false)
         {
             var now = DateTime.UtcNow;
 
             //Look for an active project
             var project = await _dbContext.Projects.Include(a => a.AcreageRate)
-                .Where(a => a.IsActive && a.Status == Project.Statuses.Active && a.Id == projectId).SingleOrDefaultAsync();
+                .Where(a =>
+                    a.IsActive &&
+                    (a.Status == Project.Statuses.Active || 
+                        (manualOverride && a.Status == Project.Statuses.AwaitingCloseout)) &&
+                    a.Id == projectId).SingleOrDefaultAsync();
             if (project == null)
             {
-                return Result.Error("No active project found for given projectId");
+                return Result.Error("No active project found for given projectId: {projectId}", projectId);
             }
 
             if (!manualOverride)
             {
-                //Make sure we are running on a business day
-                var day = now.ToPacificTime().DayOfWeek;
-                if (day == DayOfWeek.Saturday || day == DayOfWeek.Sunday)
+                if (!now.IsBusinessDay())
                 {
-                    return Result.Error("Invoices can only be created Monday through Friday");
+                    return Result.Error("Invoices can only be created Monday through Friday: {projectId}", projectId);
                 }
 
                 //Check to see if there is an invoice within current month
                 if (await _dbContext.Invoices.AnyAsync(a => a.ProjectId == projectId && a.CreatedOn.Year == DateTime.UtcNow.Year && a.CreatedOn.Month == DateTime.UtcNow.Month))
                 {
-                    return Result.Error("An invoice already exists for current month");
+                    return Result.Error("An invoice already exists for current month: {projectId}", projectId);
                 }
 
                 if (project.Start.AddMonths(1) > now) //Start doing invoices 1 month after the project starts
                 {
-                    return Result.Error("Project has not yet started");
+                    return Result.Error("Project has not yet started: {projectId}", projectId);
                 }
 
-            }
-
-            //TODO: Review this later
-            if (project.End < now)
-            {
-                //Ok, so the project has ended, we should probably create the last invoice, and close it out. Setting the project to completed.
-                project.Status = Project.Statuses.Completed;
-            }
+                //Don't create an invoice for project past its end date, and just mark it as AwaitingCloseout
+                if (project.End < now)
+                {
+                    project.Status = Project.Statuses.AwaitingCloseout;
+                    await _dbContext.SaveChangesAsync();

```

Comment:
Should we generate an email at this time to the FM that this project is awaiting closeout?

Reply:
Maybe? But I think that should be a separate issue.